ports/espressif/common-hal/socketpool/Socket.c

```diff
@@ -577,3 +599,18 @@ bool common_hal_socketpool_writable(socketpool_socket_obj_t *self) {
     // including returning true in the error case
     return num_triggered != 0;
 }
+
+void socketpool_socket_move(socketpool_socket_obj_t *self, socketpool_socket_obj_t *sock) {
+    *sock = *self;
+    self->connected = false;
+    self->num = -1;
+}
+
+void socketpool_socket_reset(socketpool_socket_obj_t *self) {
```

Comment:
does this need to close the underlying socket, or is the invariant that the underlying socket is already always closed?

Reply:
It should have never been used. This is really to set the memory so it reads as closed before it is used. Happy to rename if you have a better name suggestion.